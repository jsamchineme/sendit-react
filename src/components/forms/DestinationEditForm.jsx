import React from 'react'
import { Field, reduxForm } from 'redux-form';
import { editDestinationRules } from '../forms/rules/formRules';

export const renderField = ({
  input,
  label,
  type,
  className,
  placeholder,
  meta: { touched, error, warning }
}) => {

  let inputBox = <input className={className} {...input} placeholder={placeholder} type={type} />;
  return (
    <div>
      {touched &&
        ((error && <div className="error-box active">{error}</div>))}
        {inputBox}
    </div>
  )
}

export const Form = ({
  handleSubmit
}) => (
  <form onSubmit={handleSubmit}>
    <Field
      name="to"
      type="text"
      className="line-input"
      component={renderField}
      label="Delivery Location"
      placeholder="Provide delivery location"
      validate={editDestinationRules.to}
    />
  </form>
)

const EditForm = reduxForm({
  form: 'destinationEdit',
})(Form);

export const DestinationEditContainer = props => {
  const { handleSubmit, value } = props;
  return (
    <EditForm
      initialValues={{to: value }}
      enableReinitialize={true}
    />
  );
}

export default DestinationEditContainer;

