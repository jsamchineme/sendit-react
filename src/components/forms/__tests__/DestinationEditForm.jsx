import React from 'react';
import { shallow } from 'enzyme';
import { DestinationEditContainer } from '../DestinationEditForm';
import configureStore from '../../../redux/configureStore';

const store = configureStore();

describe('Tests for destination edit form', () => {
  it('should render the destination edit form', () => {
    const props = {
      handleSubmit: jest.fn(), 
      serverErrorMessage: '',
    }
    const wrapper = shallow(<DestinationEditContainer  {...props} />);
  
    expect(wrapper).toMatchSnapshot();
  });
});