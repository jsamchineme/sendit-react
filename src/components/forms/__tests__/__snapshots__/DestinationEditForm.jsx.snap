// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for destination edit form should render the destination edit form 1`] = `
<ReduxForm
  destroyOnUnmount={true}
  enableReinitialize={true}
  forceUnregisterOnUnmount={false}
  form="destinationEdit"
  getFormState={[Function]}
  initialValues={
    Object {
      "to": undefined,
    }
  }
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
/>
`;
